File: frontend/src/app/shared/responses/multiple-posts.response.ts
export class MultiplePostsResponse {
  postCount: number;
  page: number;
  postDetails: PostDetails[];

  constructor(postCount: number, page: number, postDetails: PostDetails[]) {
    this.postCount = postCount;
    this.page = page;
    this.postDetails = postDetails;
  }
}

export class PostDetails {
  id: number;
  name: string;
  youtubeLink: string;
  date: string;
  content: string;
  active: boolean;
  coverImageId: number;

  constructor(id: number, name: string, youtubeLink: string, date: string, content: string, active: boolean, coverImageId: number) {
    this.id = id;
    this.name = name;
    this.youtubeLink = youtubeLink;
    this.date = date;
    this.content = content;
    this.active = active;
    this.coverImageId = coverImageId;
  }
}

export class PostDetailsWithImage {
  id: number;
  name: string;
  youtubeLink: string;
  date: string;
  content: string;
  active: boolean;
  coverImage: string;

  constructor(id: number, name: string, youtubeLink: string, date: string, content: string, active: boolean, coverImage: string) {
    this.id = id;
    this.name = name;
    this.youtubeLink = youtubeLink;
    this.date = date;
    this.content = content;
    this.active = active;
    this.coverImage = coverImage;
  }
}

File: src/main/java/com/sgorokh/TimeSaver/controllers/responses/MultiplePostsResponse.java
package com.sgorokh.TimeSaver.controllers.responses;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;

import java.util.Date;
import java.util.List;

@Builder
@Data
@AllArgsConstructor
public class MultiplePostsResponse {

    private long postCount;
    private int page;
    private List<PostDetails> postDetails;

    @Builder
    @Data
    public static class PostDetails {
        private Long id;
        private String name;
        private String youtubeLink;
        private String date;
        private String content;
        private Boolean active;
        private Long coverImageId;
    }

}
